cardano-c/src/address.rs

```diff
@@ -36,9 +36,9 @@ pub extern "C" fn cardano_address_is_valid(c_address: *mut c_char) -> c_int {
 }
 
 #[no_mangle]
-pub extern "C" fn cardano_address_new_from_pubkey(c_xpubkey: XPubPtr) -> AddressPtr {
+pub extern "C" fn cardano_address_new_from_pubkey(c_xpubkey: XPubPtr, protocol_magic: ProtocolMagic) -> AddressPtr {
     let xpub = unsafe { c_xpubkey.as_ref() }.expect("Not a NULL PTR");
-    let ea = ExtendedAddr::new_simple(xpub.clone());
+    let ea = ExtendedAddr::new_simple(xpub.clone(), NetworkMagic::from(protocol_magic));
```

Comment:
no need for this PR. just out of curiosity or knowledge sharing. Did you know you could use `Into` for everytime you have `From` implemented ?

This allows you to write things like:

```rust
    let ea = ExtendedAddr::new_simple(xpub.clone(), protocol_magic.into());
```

I like the verbose way, less chance to get it wrong. Still less character too.